src/services/user.service.js

```diff
@@ -49,6 +50,55 @@ export async function createUser(fields){
     return user
 }
 
-export async function updateUser(evt){
-    let connection = await getConnection()
-}
\ No newline at end of file
+/*
+export const updateUser = async (queryParams, updateParams) => {
+    const userRepos = await getUserRepository()
+    let user = null
+    if(typeof queryParams === 'number' || typeof queryParams === 'bigint') {
+        user = await userRepos.findOne(queryParams)
+    } else {
+        user = await userRepos.find(queryParams)
+    }
+
+    for (const [key, value] of Object.entries(updateParams)) {
+        user[key] = value
+    }
+    
+    return userRepos.save(user)
+}
+*/
+
+export const userService = ({}) => {
+    const getUserById = async (id) => {
+        const rep = await getUserRepository()
+        return rep.findOne({id: id})
+    }
+
+    /**
+     * Update user
+     * 
+     * @param {Object|string|int|boolean} queryParams Query parameters in typeorm style 
+     * @param {Object} updateParams Update parameters in typeorm style
+     * @returns {Object} User
+     */
+    const updateUser = async (queryParams, updateParams) => {
+        const userRepos = await getUserRepository()
+        let user = null
+        if(typeof queryParams === 'number' || typeof queryParams === 'bigint') {
+            user = await userRepos.findOne(queryParams)
+        } else {
+            user = await userRepos.find(queryParams)
+        }
+
+        for (const [key, value] of Object.entries(updateParams)) {
```

Comment:
Minor on this one. In theory this can be rewitten as
```
user = {...user, ...updateParams} 
```
Unless the user has properties that cannot be spread.  🤔 

In the scope of this `userRepos.save(user)` might also need verifying just to check that bad data cannot be directly injected into it. Though I think we can ignore that for now given we don't directly run any update operations against the API as of now